src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -667,7 +667,9 @@ void reportDanglingDocComment(Comment c) {
         var pos = c.getPos();
         if (pos != null) {
             deferredLintHandler.report(lint -> {
-                if (lint.isEnabled(Lint.LintCategory.DANGLING_DOC_COMMENTS)) {
+                if (lint.isEnabled(Lint.LintCategory.DANGLING_DOC_COMMENTS) &&
+                        (c.getStyle() != Comment.CommentStyle.JAVADOC_LINE ||
+                                c.getPos().getStartPosition() != 0)) {
```

Comment:
For consideration, you could further refine the test by making sure it is a single line comment (perhaps optionally terminated by a newline).  In other words, maybe the warning should be given if the leading comment is a multi-line comment -- which would unlikely be a JBang-style comment.

Reply:
I admit, I thought about it initially, but couldn't quickly come up with a simple and robust way to figure out if a comment tree spans a single line. I'm not yet comfortable working with compiler coordinate system (incl. source positions). If you propose such a way, I'd happily incorporate it.

Reply:
> I admit, I thought about it initially, but couldn't quickly come up with a simple and robust way to figure out if a comment tree spans a single line. I'm not yet comfortable working with compiler coordinate system (incl. source positions). If you propose such a way, I'd happily incorporate it.

Does the content of the comment (`c.getText()`) contain a newline character that is not at the end of the content?  Maybe something like 
  `c.getText().matches("(?s)\\R.")` 
although there may be more effective ways to do that.

Reply:
Regex sounds simple and robust. However, I thought we could concoct a more idiomatic compiler solution, using trees, start and end diagnostic positions, and also line numbers.

So, if a comment tree start and end position are on the same line that is also the first line of the file, we don't output the warning. Can we do that?

Reply:
Computing line numbers is a relatively expensive operation, because the internal coordinates are character-offset based. That being said, there is a bunch of caching going on under the covers, and, the check can be guarded by the additional (proposed) check that the check is only suppressed if the comment begins at offset 0.

Reply:
> So, if a comment tree start and end position are on the same line that is also the first line of the file

The end position will always be after the start position, so you only need check if the line number of the end position of the comment is line 1. (Line numbers are 1-based.)

Reply:
Jon, please have a look at the updated version and approve if it is to your liking; thanks.